.github/workflows/release.yml

```diff
@@ -0,0 +1,47 @@
+name: Release
+
+on:
+  release:
+    types:
+      - published
+
+env:
+  DEFAULT_PYTHON: "3.11"
+
+permissions:
+  contents: read
+
+jobs:
+  release-pypi:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi-dev
+      url: https://test.pypi.org/project/pydocstyle/
+    environment:
+      name: pypi-prod
+      url: https://pypi.org/project/pydocstyle/
+    steps:
+      - name: Check out code from Github
+        uses: actions/checkout@v3.2.0
+      - name: Set up Python ${{ env.DEFAULT_PYTHON }}
+        id: python
+        uses: actions/setup-python@v4.4.0
+        with:
+          python-version: ${{ env.DEFAULT_PYTHON }}
+          check-latest: true
+      - name: Install requirements
+        run: |
+          python -m pip install -U pip twine wheel
+          python -m pip install -U "setuptools>=56.0.0"
+      - name: Build distributions
+        run: |
+          python setup.py sdist bdist_wheel
```

Comment:
Direct invocation of `setup.py` is [deprecated](https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html), current [best practice](https://packaging.python.org/en/latest/tutorials/packaging-projects/#generating-distribution-archives) is to use `python -m build` instead.